File: app/src/types/paciente.ts
import { StatusPaciente } from "../enums/statusPaciente";

export interface Paciente {
  nome: string;
  id: number;
  possuiEvolucaoHoje: boolean;
  status: StatusPaciente;
}

export interface PacienteForm {
  nome: string;
  remediosUsoContinuo?: string;
  pontosAtencao?: string;
}

export interface Resumo {
  id: number;
  nome: string;
  status: StatusPaciente;
  pontosAtencao?: string;
  remediosUsoContinuo?: string;
  graficoHumor: GraficoHumor[];
  graficoComportamento: GraficoComportamento[];
}

export interface GraficoHumor {
  data: string;
  valor: number;
}

export interface GraficoComportamento {
  nome: string;
  quantidade: number;
}

File: api/aceso/src/main/java/com/tcc/aceso/api/domain/Paciente.java
package com.tcc.aceso.api.domain;

import com.tcc.aceso.api.enums.StatusPaciente;
import jakarta.persistence.*;
import lombok.*;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "pacientes")
public class Paciente {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String nome;

    @Column(name = "remedios_uso_continuo", columnDefinition = "TEXT")
    private String remediosUsoContinuo;

    @Column(name = "pontos_atencao", columnDefinition = "TEXT")
    private String pontosAtencao;

    @Column(nullable = false)
    @Enumerated(EnumType.STRING)
    private StatusPaciente status = StatusPaciente.ATIVO;
}
